backtracking/remove_invalid_parenthese.py

```diff
@@ -0,0 +1,120 @@
+def remove_invalid_parentheses(input_string: str) -> list[str]:
+    """
+    Removes the minimum number of invalid parentheses
+    to make the input string valid.
+
+    Args:
+        input_string (str): The input string containing parentheses.
+
+    Returns:
+        List[str]: A list of valid strings after removing
+        the minimum number of parentheses.
+
+    Examples:
+        >>> remove_invalid_parentheses(")(a)())")
+        ['(a())', '(a)()']
+
+        >>> remove_invalid_parentheses("()())()")
+        ['(())()', '()()()']
+
+        >>> remove_invalid_parentheses(")(")
+        ['']
+    """
+
+    def is_valid(next_string: str) -> bool:
+        """
+        Helper function to check if a string has valid parentheses.
+
+        Params:
+            next_string (str): The input string to be checked.
+        Returns:
+            bool: True if the parentheses in the string are valid, False otherwise.
+
+        Examples:
+        >>> is_valid("(a)())")
+        False
+
+        >>> is_valid("(a())")
+        True
+
+        >>> is_valid(")(a)()")
+        False
+
+        """
+        count: int = 0
+        for char in next_string:
+            if char == "(":
+                count += 1
+            elif char == ")":
+                count -= 1
+                if count < 0:
+                    return False
+        return count == 0
+
+    def dfs(
+        current_string: str, start_index: int, left_removed: int, right_removed: int
+    ) -> None:
+        """
+        Depth-first search function to generate valid combinations
+        of the input string.
+
+        Params:
+            current_string (str): The current string being processed.
+            start_index (int): The index to start processing from.
+            left_removed (int): The number of left parentheses to be removed.
+            right_removed (int): The number of right parentheses to be removed.
+        Returns:
+            None
+        Examples:
+        >>> dfs("()())()", 0, 0, 1)
+        None
+
+        >>> dfs("()())(", 6, 0, 0)
+        None
+
+        >>> dfs(")(a)())", 0, 0,, 2)
+        None
```

Comment:
These tests only prove that the function ran.  We need more,

Reply:
but its only require to search and make changes into the outer declared variable, it doesnt returns value.